yarpl/src/yarpl/utils/credits.cpp

```diff
@@ -89,6 +89,28 @@ int64_t consume(std::atomic<std::int64_t>* current, int64_t n) {
   }
 }
 
+bool tryConsume(std::atomic<std::int64_t>* current, int64_t n) {
```

Comment:
we should 100% make this class single-threaded, and have all threadedness be explicit in the codebase. It's unclear what objects and methods are alright to call from multiple threads, and what needs explicit synchronization.